qiskit/algorithms/minimum_eigen_solvers/qaoa/qaoa.py

```diff
@@ -127,34 +119,30 @@ def __init__(self,
         validate_min('p', p, 1)
 
         self._p = p
-        self._mixer = mixer.to_opflow() if isinstance(mixer, LegacyBaseOperator) else mixer
+        self._mixer = mixer
         self._initial_state = initial_state
 
         # VQE will use the operator setter, during its constructor, which is overridden below and
         # will cause the var form to be built
-        super().__init__(operator,
-                         None,
-                         optimizer,
+        super().__init__(var_form=None,
+                         optimizer=optimizer,
                          initial_point=initial_point,
                          gradient=gradient,
                          expectation=expectation,
                          include_custom=include_custom,
                          max_evals_grouped=max_evals_grouped,
                          callback=callback,
-                         quantum_instance=quantum_instance,
-                         aux_operators=aux_operators)
+                         quantum_instance=quantum_instance)
 
-    @VQE.operator.setter  # type: ignore
-    def operator(self, operator: Union[OperatorBase, LegacyBaseOperator]) -> None:
-        """ Sets operator """
+    def _check_operator(self,
+                        operator: OperatorBase) -> OperatorBase:
         # Need to wipe the var_form in case number of qubits differs from operator.
```

Comment:
I guess this comment is kinda valid. It used to be more precise since the var form variable was set to None i.e. wiped. Now it seems the var form is simply re-built each time this is called.

Reply:
I rewrote the comment since var_form is always recreated with parameter operator